src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -114,17 +114,15 @@ public void EndEvent(object eventHandle)
                             Data = evt.Data
                         };
                     },
-                    (name, evtToUpdate) =>
+                    (name, existing) =>
```

Comment:
You're pointing out that the update delegate isn't run under a lock and can therefore run concurrently. Does the change below actually improve things? E.g. in a case where 2 updates run concurrently, your code will create a new value from the "existing" value passed in, but they'll both get the same existing value, so it's still last one wins?

Reply:
This is correct. After the change, when 2 updates are run concurrently, only 1 will succeed in this loop https://github.com/dotnet/runtime/blob/12299e2ed8e926ce20c3af6f49de39ee0d17e71c/src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs#L1253-L1272 and the other will spin again, recalculate the value and try to update again. This is change is about correctness in reporting, not performance (for sure it will alloc at least one object per update).

Reply:
I'm going to run these changes using our [Crank](https://github.com/dotnet/crank) setup to measure the perf impact.

Reply:
👍 For sure there will be some, this allocates at least one instance per `AddOrUpdate`. 
If we want to make it really fast and correct at the same time, I'll need to kill this PR and issue another one. I think this would result in a bigger set of changes. Anyway, looking forward to hearing more!